### PR TITLE
Add support for CreateEventPayload and DeleteEvent payload (#1646)

### PR DESCRIPTION
**Pull request by GMouron**
Hello, I tried to do the same as was done in #1732 
Please tell me if anything is missing.
Cheers

### CODE REVIEW

**Comment by ryangribble** [comment]
Hi @GMouron, thanks for this!  It looks good although one thing I noticed when checking the docs is that the create event payload has  `master_branch` and `description` fields as well, so it would seem a good idea to add them to the response model in this PR.  What do you think?
https://developer.github.com/v3/activity/events/types/#createevent

File: Octokit/Models/Response/RefType.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        /// <summary>
+        /// The object is of type repository
+        /// </summary>
+        [Parameter(Value = "repository")]
+        Repository,
+        /// <summary>
+        /// The object is of type branch
+        /// </summary>
+        [Parameter(Value = "branch")]
```
Comment (ryangribble) [inline Octokit/Models/Response/RefType.cs]: minor nit: missing a whitespace line

Comment (GMouron) [reply Octokit/Models/Response/RefType.cs]: fixed :)

**Comment by ryangribble** [comment]
>> as well as the content of the events sent by github for the creation of a branch which does not contain them, I've decided to omit these here since, from my understanding, they will never be part of the payload we receive

Octokit.net uses these payload response classes for the results from "Activity Event" Api, which does include repository events and thus does include these fields.

The response classes can also be used to deserialise webhook payloads, although we dont natively have support for webhooks, so users doing this would be having to call the SipleJsonSerializer themselves currently.

So I think the extra fields should be included, since the current main supported use case in Octokit (the events API) would use them.  They would then just be `null` for anyone using them from a webhook perspective.

When we implement native webhook support we can consider whether to re-use these payload classes or have new ones specifically for the webhook payloads

**Comment by GMouron** [comment]
OK, I'll be adding them then :)

**Comment by GMouron** [comment]
And done :)

**Comment by ryangribble** [comment]
release_notes: Support `CreateEvent` and `DeleteEvent` payloads, using new response models `CreateEventPayload` and `DeleteEventPayload`

**Comment by ryangribble** [comment]
Thanks @GMouron!  (we have quite a few payload events that aren't implemented - it would be great if you wanted to add more!)

LGTM

**Comment by GMouron** [comment]
> Thanks @GMouron! (we have quite a few payload events that aren't implemented - it would be great if you wanted to add more!)
> 
> LGTM

Hi, I might take a look into that a bit later but I really needed those ones and I have to first take care of the code that depends on them 😄

**Pull request merged**